source-mongodb/main.go

```diff
@@ -93,6 +100,16 @@ func (d *driver) Connect(ctx context.Context, cfg config) (*mongo.Client, error)
 		return nil, err
 	}
 
+	if diff, err := OplogTimeDifference(ctx, client); err != nil {
```

Comment:
> I'm assuming that wouldn't work here since we're not exactly checking for the actual oplog retention period, and rather the inferred retention period from the oldest and newest records - and that could be skewed if there are very infrequent updates. Curious if you have any other thoughts on this though.

Yep this is the issue, we can't check an actual retention period for Mongo versions prior to 4.4, and after 4.4 it is an optional parameter that can be set: https://www.mongodb.com/docs/manual/core/replica-set-oplog/#minimum-oplog-retention-period

I don't really know if we need to support Mongo < 4.4, I've assumed the answer is yes. If the answer is no, this is simpler because we can just ask the user to set [storage.oplogMinRetentionHours](https://www.mongodb.com/docs/manual/reference/configuration-options/#mongodb-setting-storage.oplogMinRetentionHours) and we can check its value to make sure it is sufficient. Perhaps we can have both implementations if we want to support < 4.4 versions.